numpy/core/src/multiarray/convert.c

```diff
@@ -46,7 +46,7 @@ npy_fallocate(npy_intp nbytes, FILE * fp)
     }
     /* btrfs can take a while to allocate making release worthwhile */
     NPY_BEGIN_ALLOW_THREADS;
-    r = fallocate(fileno(fp), 0, npy_ftell(fp), nbytes);
+    r = fallocate(fileno(fp), 1, npy_ftell(fp), nbytes);
```

Comment:
should have a comment that this is FALLOC_FL_KEEP_SIZE needed for the case of files opened in append mode